src/unity/toolkits/tcmps/mps_graph_cnnmodule.mm

```diff
@@ -151,35 +173,36 @@ - (instancetype)initWithCommandBuffer:(id<MTLCommandBuffer>)commandBuffer {
     [image synchronizeOnCommandBuffer:cb];
   }
 
+  // Schedule copying of the output into a float_array promise.
+  // n.b. The block below must not capture `this`.
```

Comment:
Ah yes, this comment reaffirms what I thought. Oh, and the same comment is in Train too already, I just missed it. OK, all good here! :+1:

Reply:
> TCMPS is becoming really cool! 

👍 👍   
@nickjong for the rescue! 👏